src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -317,6 +317,62 @@ public Invocation TransformInvocationVersionOne(InvocationVersionOne node)
             return invocation;
         }
 
+        public Notification TransformNotificationVersionOne(NotificationVersionOne node)
+        {
+            Notification notification = null;
+
+            if (node != null)
+            {
+                notification = new Notification
+                {
+                    Exception = TransformExceptionDataVersionOne(node.Exception),
+                    Id = node.Id,
+                    Level = node.Level.ToNotificationLevelV2(),
+                    Properties = node.Properties,
+                    RuleId = node.RuleId,
+                    RuleKey = node.RuleKey,
+                    ThreadId = node.ThreadId,
+                    Time = node.Time
+                };
+
+                if (!string.IsNullOrWhiteSpace(node.Message))
+                {
+                    notification.Message = new Message
+                    {
+                        Text = node.Message
+                    };
+                }
+            }
+
+            return notification;
+        }
+
+        public ExceptionData TransformExceptionDataVersionOne(ExceptionDataVersionOne node)
+        {
+            ExceptionData exceptionData = null;
+
+            if (node != null)
+            {
+                exceptionData = new ExceptionData
+                {
+                    Kind = node.Kind,
+                    Message = node.Message
+                };
+
+                if (node.InnerExceptions != null)
+                {
+                    exceptionData.InnerExceptions = new List<ExceptionData>();
+
+                    foreach (ExceptionDataVersionOne edvo in node.InnerExceptions)
+                    {
+                        exceptionData.InnerExceptions.Add(TransformExceptionDataVersionOne(edvo));
+                    }
+                }
+            }
+
+            return exceptionData;
+        }
+
         public override RunVersionOne VisitRunVersionOne(RunVersionOne node)
```

Comment:
This method is getting big. I suggest refactoring as follows:
```
public override RunVersionOne VisitRunVersionOne(RunVersionOne node)
{
    run.Invocations = new List<Invocation>();
    run.Invocations.Add(
        CreateInvocation(node.Invocation, node.ToolInvocations, node.ConfigurationNotifications);
    ...
}

Invocation CreateInvocation(
    InvocationVersionOne v1Invocations,
    IEnumerable<NotificationVersionOne> v1ToolNotifications,
    IEnumerable<NotificationVersionOne> v1ConfigurationNotifications)
{
    Invocation invocation = v1Invocation != null
        ? TransformInvocationVersionOne(v1Invocation)
        : new Invocation();

    invocation.ToolNotifications = TransformNotificationsVersionOne(v1ToolNotifications);
    invocation.ConfigurationNotifications = TransformNotificationsVersionOne(v1ConfigurationNotifications);

    return invocation;
}

public IList<Notification> TransformNotificationsVersionOne(IEnumerable<NotificationVersionOne v1Notifications)
{
    var notification = new List<Notification>();
    if (v1Notifications != null)
    {
        foreach (NotificationVersionOne configNotification in node.ConfigurationNotifications)
        {
            notification.Add(TransformNotificationVersionOne(configNotification));
        }
    }

    return notifications;
}

public Notification TransformNotificationVersionOne(NotificationVersionOne v1Notification)
{
    ...
}
```

Reply:
One issue, if we have no invocation and no notifications, we'll end up with an empty invocation